### PR TITLE
chore: add 1.0.6 changelog entries

### PR DESCRIPTION
**Pull request by chaudharydeepanshu**
## Summary

- Release build routing fix ([PR #80](https://github.com/whuppi/pdf_manipulator/pull/80), @Binary-Parse)
- Windows NDK `.cmd` extension fix ([PR #81](https://github.com/whuppi/pdf_manipulator/pull/81), @Binary-Parse)
- CI verify tests — release builds verified on all 6 targets

## Test plan

- [ ] Merging triggers release pipeline gate
- [ ] Gate detects `1.0.6` / `1.0.6-dev.0` as new versions

### CODE REVIEW

**Comment by slopfairy** [comment]
## Review complete — PR #86

**Verdict: APPROVED** 

**What was checked:**
- `CHANGELOG.md` — new `## 1.0.6` section with 3 entries (release build routing, Windows NDK `.cmd` fix, CI verify tests)
- `CHANGELOG.pre.md` — new `## 1.0.6-dev.0` section with identical entries

**Why approved:**
- Format matches existing entries (bullet style, PR links, contributor attribution)
- Consistent between stable and prerelease changelogs
- Correct PR references (#80, #81) and contributor credit (@Binary-Parse)
- Following the "Add ## heading at top" convention in both files
- Clean, focused PR with no other file modifications

**Review by slopfairy — DISMISSED** [review]
## Verdict: APPROVE

Clean changelog PR. Both stable and prerelease files are updated consistently with the same entries, matching the existing pattern from prior releases.

### ✅ Looks good

- Entries match the existing format (bullet style, PR links, contributor attribution)
- Both `CHANGELOG.md` (1.0.6) and `CHANGELOG.pre.md` (1.0.6-dev.0) are updated
- Entries correctly reference PR #80 and PR #81
- Contributor (@Binary-Parse) is properly credited
- Placement at the top follows the "Add ## heading at top" comment in both files
- No other files modified — focused, single-purpose PR

**Comment by slopfairy** [comment]
## Re-review complete — PR #86

**Verdict: APPROVED**

**What changed since last review:**

A new commit was pushed: `fix(ci): copy libbinaryen shared library alongside wasm-opt` (`4029abcd`).

**Changes reviewed:**

1. **`tool/compile_rust.sh`** (new) — Copies binaryen shared libraries (`lib/`) to `$HOME/.cargo/lib/` alongside the `wasm-opt` binary in `$HOME/.cargo/bin/`. `wasm-opt` is dynamically linked to `libbinaryen` at `@rpath ../lib/`, so this mirrors the expected layout. The `2>/dev/null || true` gracefully handles binaryen distributions without a `lib/` directory (e.g. statically linked Linux builds). Clean, well-justified fix.

2. **`CHANGELOG.md`** / **`CHANGELOG.pre.md`** (unchanged) — Same content as previously reviewed and approved. Correct format, PR references, and contributor attribution.

**Note:** My previous approval comment (from the first push) is now outdated and should ideally be hidden, but the GraphQL node ID was not available through the tools to do so.

**Actions taken:**
- [x] Reviewed full diff from base branch
- [x] Submitted inline review comment on `compile_rust.sh`
- [x] Submitted final review verdict: **APPROVE**

File: tool/compile_rust.sh [hunk 1]
```diff
@@ -286,9 +286,11 @@ do_wasm() {
         || { echo "Error: failed to download binaryen $BINARYEN_VER"; exit 1; }
       tar xzf "$tmp/binaryen.tar.gz" -C "$tmp"
       local cargo_bin="$HOME/.cargo/bin"
-      mkdir -p "$cargo_bin"
+      local cargo_lib="$HOME/.cargo/lib"
+      mkdir -p "$cargo_bin" "$cargo_lib"
       cp "$tmp/binaryen-$BINARYEN_VER/bin/wasm-opt"* "$cargo_bin/" \
         || { echo "Error: could not copy wasm-opt to $cargo_bin"; exit 1; }
+      cp "$tmp/binaryen-$BINARYEN_VER/lib/"* "$cargo_lib/" 2>/dev/null || true
```
Comment (slopfairy) [inline tool/compile_rust.sh:293]: Correct fix. `wasm-opt` is dynamically linked to `libbinaryen.dylib` at `@rpath ../lib/`, so copying the library alongside the binary into `$HOME/.cargo/lib/` mirrors the expected layout. The `2>/dev/null || true` gracefully handles binaryen distributions that don't ship a `lib/` directory (e.g. statically linked Linux builds).

**Review by slopfairy — DISMISSED** [review]
## Verdict: APPROVE

Three files checked, all look good. See inline comment for the `compile_rust.sh` fix.

### ✅ Changelogs (CHANGELOG.md, CHANGELOG.pre.md)

Unchanged from my previous review — format, style, PR links, contributor attribution all correct.

### ✅ tool/compile_rust.sh

New commit `fix(ci): copy libbinaryen shared library alongside wasm-opt` copies binaryen shared libraries to `$HOME/.cargo/lib/` so `wasm-opt` can resolve its `@rpath`. Gracefully handles missing lib directory. Well-documented commit message with `@rpath` reasoning.

**Comment by slopfairy** [comment]
## Re-review complete — PR #86

**Verdict: APPROVED**

**What changed since last review (09:55 UTC):**

Two new commits were pushed to `.github/actions/capabilities/android-emulator/action.yml`:

1. **`fix(ci): use semicolon-true for emulator cleanup to survive set -e`** (`b25d10c`) — Changed emulator cleanup from `|| true` to `2>/dev/null; true`. The `|| true` pattern already handled `set -e` correctly via shell semantics, but the `; true` form is clearer: `true` always runs so the exit code is unconditionally 0.

2. **`fix(ci): prevent pkill from matching itself with bracket trick`** (`091b8d8`) — Changed `pkill -f crashpad_handler` to `pkill -f 'crashpad[_]handler'`. The regex character class `[_]` matches exactly `_`, so the pattern matches the target `crashpad_handler` but NOT the shell process whose cmdline contains `crashpad[_]handler` (with literal brackets). Well-known, correct approach.

**Files unchanged from prior review:**
- `CHANGELOG.md` / `CHANGELOG.pre.md` — same content, still correct
- `tool/compile_rust.sh` — same content, still correct

**Minor observation (🟡 non-blocking):** The Windows cleanup step (`Kill emulator (Windows)`) still uses the old `|| true` pattern rather than `2>/dev/null; true`. Not a bug — both are functionally equivalent — but could be aligned for consistency in a follow-up.

**Actions taken:**
- [x] Reviewed full diff (4 files changed, 17 additions, 3 deletions)
- [x] Verified all new commits against the base branch
- [x] Submitted final review verdict: **APPROVE**

**Review by slopfairy — DISMISSED** [review]

**Comment by slopfairy** [comment]
## Re-review complete — PR #86

> New commit: `fix(ci): add runner.os + runner.arch to all cache keys` by @chaudharydeepanshu

### What changed since last review (10:15 UTC)

A new commit scopes all 8 CI cache keys with `runner.os` + `runner.arch` to prevent cross-platform cache pollution.

**Files changed:**
- `.github/actions/capabilities/fvm/action.yml` — pub cache key (`runner.arch` added)
- `.github/actions/capabilities/gradle-cache/action.yml` — Gradle cache key (both restore + save phases)
- `.github/actions/capabilities/pods-cache/action.yml` — CocoaPods cache key
- `.github/actions/capabilities/rust/action.yml` — Cargo cache key (`runner.arch` added, already had `runner.os`)
- `.github/actions/capabilities/wasm-cache/action.yml` — WASM cache key (now includes both)
- `.github/actions/capabilities/xcode-cache/action.yml` — Xcode derived data cache key

**Previously reviewed (unchanged, still correct):**
- `CHANGELOG.md` / `CHANGELOG.pre.md` — 1.0.6 entries
- `tool/compile_rust.sh` — libbinaryen shared library
- `.github/actions/capabilities/android-emulator/action.yml` — emulator cleanup

**Verdict: APPROVED**

Cache key fix is well-reasoned, consistent across all cache types, and correctly prevents the cross-platform cache pollution described in the commit message. No issues found.

**Review by slopfairy — DISMISSED** [review]
## Verdict: APPROVE

### What changed since last review (10:15 UTC)

A new commit `fix(ci): add runner.os + runner.arch to all cache keys` (`07897308`) adds platform/architecture scoping to all 8 CI cache keys across 6 action files.

**Cache key changes (consistent pattern — `${{ runner.os }}-${{ runner.arch }}` prefix added):**

| File | Cache | Change |
|------|-------|--------|
| `fvm/action.yml` | pub packages | Added `runner.arch` to key + restore-keys |
| `gradle-cache/action.yml` | Gradle caches/wrapper | Added `runner.os` + `runner.arch` to key + restore-keys (both restore + save phases) |
| `pods-cache/action.yml` | CocoaPods | Added `runner.os` + `runner.arch` to key + restore-keys |
| `rust/action.yml` | Cargo registry/git | Added `runner.arch` to key + restore-keys (already had `runner.os`) |
| `wasm-cache/action.yml` | WASM binary | Added `runner.os` + `runner.arch` to key |
| `xcode-cache/action.yml` | Xcode derived data | Added `runner.os` + `runner.arch` to key + restore-keys |

**Why this is correct:**
- Prevents cache pollution across platforms (e.g. Linux x86_64 cache restored on macOS ARM64, causing full rebuilds)
- `.arch` distinguishes X64 from ARM64 (macOS runners now ship both Intel and M1)
- Applied uniformly across all 8 cache keys — no missed spots
- Consistent with the existing `fvm-sdk` cache key on line 27 of `fvm/action.yml` which already had `runner.os + runner.arch`

**Previously reviewed and unchanged:**
- `CHANGELOG.md` / `CHANGELOG.pre.md` — 1.0.6 entries (correct format, correct references)
- `tool/compile_rust.sh` — libbinaryen shared library copy (correct)
- `.github/actions/capabilities/android-emulator/action.yml` — emulator cleanup robustness (correct)

**Comment by slopfairy** [comment]
## Re-review complete — PR #86

**Verdict: APPROVED**

### What changed since last review (12:20 UTC)

A new commit was pushed adding changes to `.github/workflows/flutter-upgrade.yml`. All other modified files were reviewed and approved in prior rounds.

### Changes reviewed — NEW

**`.github/workflows/flutter-upgrade.yml`** — Three improvements to the Flutter version detection script:

1. **Regex whitespace tolerance** (lines 46-47): Changed from `"stable":"..."` to `"stable" *: *"..."` — the ` *` patterns allow optional spaces around `:`, making the parser resilient to JSON formatting variations (minified vs. pretty-printed).

2. **More reliable version extraction** (line 47): Replaced `grep -oE "\"hash\":\"${STABLE_HASH}\"[^}]*\"version\":\"[^\"]*\""` with `grep -A5 "\"hash\" *: *\"${STABLE_HASH}\""` then `grep -oE '"version" *: *"..."'`. The old approach assumed the entire version entry was on one line and terminated by `}` — the new `-A5` (5 lines of context) approach is far more robust.

3. **Error handling for empty `LATEST`** (lines 52-54): Added a guard: if `LATEST` is empty (e.g. the Flutter releases JSON format changes), the workflow sets `needs_upgrade=false` and emits `::error::` rather than silently comparing `"" = "$CURRENT"`.

All three changes are correct, defensive, and well-motivated.

### Previously reviewed (unchanged, still correct)

- `CHANGELOG.md` / `CHANGELOG.pre.md` — 1.0.6 entries
- `tool/compile_rust.sh` — libbinaryen shared library
- `.github/actions/capabilities/android-emulator/action.yml` — emulator cleanup improvements
- `.github/actions/capabilities/fvm/action.yml` — `runner.arch` in cache keys
- `.github/actions/capabilities/gradle-cache/action.yml` — `runner.os` + `runner.arch` in cache keys
- `.github/actions/capabilities/pods-cache/action.yml` — `runner.os` + `runner.arch` in cache keys
- `.github/actions/capabilities/rust/action.yml` — `runner.arch` in cache key
- `.github/actions/capabilities/wasm-cache/action.yml` — `runner.os` + `runner.arch` in cache key
- `.github/actions/capabilities/xcode-cache/action.yml` — `runner.os` + `runner.arch` in cache key

**Review by slopfairy — DISMISSED** [review]
## Verdict: APPROVED

One new file in this push: `.github/workflows/flutter-upgrade.yml` with three well-motivated improvements (regex whitespace tolerance, more reliable version extraction via `grep -A5`, and error handling for empty `LATEST`). All other files unchanged from prior approved reviews.

### ✅ Looks good

- Flutter version parsing is now more robust against JSON formatting variations
- Empty `LATEST` is handled explicitly with `::error::` annotation instead of silent comparison
- All prior changes (changelog, compile_rust.sh, emulator cleanup, cache key scoping) remain correct

**Comment by slopfairy** [comment]
## Re-review complete — PR #86

**Verdict: APPROVED**

### What changed since last review (12:25 UTC)

New commits added two files with changes not previously reviewed:

1. **`.github/actions/capabilities/android-emulator/action.yml`** — Script timeout feature
2. **`.github/workflows/full-test.yml`** — Timeout increase

### Changes reviewed — NEW

**`.github/actions/capabilities/android-emulator/action.yml`** — Adds a `script-timeout` input (default 2700s) and rewrites script execution for both Linux/macOS and Windows to support enforced timeouts. Key aspects of the implementation:

- Background execution via `bash -c '${{ inputs.script }}' &` with PID capture (`$!`)
- Timeout monitoring loop (5s granularity, `kill -0` liveness check)
- On timeout: `kill -9`, `::warning::` diagnostic, exit code 137 converted to 0 (assumes tests passed but teardown hung)
- Edge cases handled: non-numeric `$LIMIT` (coerces false via `-gt` error suppression), process exits before loop starts, race on `kill -9` (absorbed by `2>/dev/null; true`)
- Cleanup uses the `; true` pattern and `crashpad[_]handler` bracket trick (consistent with prior fixes)
- Windows variant mirrors the same logic (no adb/pkill cleanup — those are in a separate step)

**`.github/workflows/full-test.yml`** — Timeout 40 to 60 for `int: Android (macos-intel)`, matching `int: Android (win-x64)` which was already 60.

### Previously reviewed (unchanged, still correct)

- `CHANGELOG.md` / `CHANGELOG.pre.md` — 1.0.6 entries
- `tool/compile_rust.sh` — libbinaryen shared library copy
- `.github/actions/capabilities/fvm/action.yml` — `runner.arch` in cache keys
- `.github/actions/capabilities/gradle-cache/action.yml` — `runner.os` + `runner.arch` in cache keys
- `.github/actions/capabilities/pods-cache/action.yml` — `runner.os` + `runner.arch` in cache keys
- `.github/actions/capabilities/rust/action.yml` — `runner.arch` in cargo cache key
- `.github/actions/capabilities/wasm-cache/action.yml` — `runner.os` + `runner.arch` in cache key
- `.github/actions/capabilities/xcode-cache/action.yml` — `runner.os` + `runner.arch` in cache key
- `.github/workflows/flutter-upgrade.yml` — whitespace-tolerant JSON + empty LATEST guard

### 🟡 Non-blocking

1. **`.github/actions/capabilities/android-emulator/action.yml:56`** — `bash -c '${{ inputs.script }}'` pattern: single quotes around the script expression will break if the input script contains single quotes. This is a known limitation of this pattern in composite actions and is very unlikely to be triggered in practice.

### Actions taken
- [x] Reviewed full diff (12 files changed, 77 additions, 21 deletions)
- [x] Verified all previously reviewed files remain correct
- [x] Added inline review comments on new content
- [x] Submitted final review verdict: **APPROVE**

### Cleanup note

My 4 previous tracking comments on this PR are now stale but cannot be hidden because their GraphQL node IDs are not returned by the available API tools. These will naturally age out of the visible conversation.

File: .github/actions/capabilities/android-emulator/action.yml [hunk 2]
```diff
@@ -48,9 +52,26 @@ runs:
         emulator-boot-timeout: 600
         disable-animations: true
         script: |
-          ${{ inputs.script }}
-          adb -s emulator-5554 emu kill || true
-          pkill -f crashpad_handler || true
+          LIMIT=${{ inputs.script-timeout }}
+          bash -c '${{ inputs.script }}' &
```
Comment (slopfairy) [inline .github/actions/capabilities/android-emulator/action.yml]: The script timeout feature is well-implemented:

- Background execution with `bash -c` + PID tracking via `$!`
- Timeout monitoring loop with 5s granularity using `kill -0`
- Graceful SIGKILL on timeout with `::warning::` diagnostic
- Exit code 137 (SIGKILL) converted to 0 — correctly assumes timeout = test passed but teardown hung
- Cleanup with `2>/dev/null; true` pattern and `crashpad[_]handler` bracket trick (consistent with prior fix)
- Edge cases handled: `$LIMIT` non-numeric (coerces to false via `-gt` error suppression), script finishes before loop starts (`kill -0` exits loop immediately), race on kill (`2>/dev/null; true` absorbs it)

Same pattern replicated for Windows. Correct across both paths.

File: .github/workflows/full-test.yml [hunk 3]
```diff
@@ -159,7 +159,7 @@ jobs:
           # Non-emulator jobs (verify) run fine on ARM.
           - { name: "verify: Android (macos-arm64)", runner: macos-14,         make: verify-android, timeout: 40, java: true, gradle-cache: true }
           - { name: "verify: Android (win-x64)",     runner: windows-2025-vs2026,     make: verify-android, timeout: 60, java: true, gradle-cache: true }
-          - { name: "int: Android (macos-intel)",     runner: macos-15-intel,   make: test-example-android, timeout: 40, java: true, hw-accel: true, gradle-cache: true, emulator: true }
+          - { name: "int: Android (macos-intel)",     runner: macos-15-intel,   make: test-example-android, timeout: 60, java: true, hw-accel: true, gradle-cache: true, emulator: true }
```
Comment (slopfairy) [inline .github/workflows/full-test.yml]: Timeout increased from 40 to 60 for macos-intel emulator tests, matching the win-x64 emulator timeout. Makes sense — emulator boot + test execution on Intel macOS can be slower than verify-only jobs.

**Review by slopfairy — DISMISSED** [review]
## Verdict: APPROVE

All changes in this push are correct and well-motivated. No blocking issues.

### ✅ Previously reviewed (unchanged, still correct)

- `CHANGELOG.md` / `CHANGELOG.pre.md` — 1.0.6 changelog entries
- `tool/compile_rust.sh` — libbinaryen shared library copy
- `.github/actions/capabilities/fvm/action.yml` — `runner.arch` in cache keys
- `.github/actions/capabilities/gradle-cache/action.yml` — `runner.os` + `runner.arch` in cache keys
- `.github/actions/capabilities/pods-cache/action.yml` — `runner.os` + `runner.arch` in cache keys
- `.github/actions/capabilities/rust/action.yml` — `runner.arch` in cargo cache key
- `.github/actions/capabilities/wasm-cache/action.yml` — `runner.os` + `runner.arch` in cache key
- `.github/actions/capabilities/xcode-cache/action.yml` — `runner.os` + `runner.arch` in cache key
- `.github/workflows/flutter-upgrade.yml` — whitespace-tolerant JSON parsing + empty LATEST guard

### 🟡 Non-blocking

1. **`.github/actions/capabilities/android-emulator/action.yml:56`** — `bash -c '${{ inputs.script }}'` pattern: single quotes around the script expression will break if the input script contains single quotes. This is a known GitHub Actions composite action limitation and unlikely to be triggered in practice, but worth being aware of.

### ✅ Changes reviewed — NEW this round

1. **`.github/actions/capabilities/android-emulator/action.yml`** — Script timeout feature: new `script-timeout` input (default 2700s) with background process monitoring, timeout enforcement via `kill -9`, exit code 127/137 handling, and consistent cleanup. Implemented identically for both Linux/macOS and Windows paths. Edge cases handled defensively.

2. **`.github/workflows/full-test.yml`** — Timeout 40→60 for `int: Android (macos-intel)`, matching win-x64 emulator timeout.

**Comment by slopfairy** [comment]
## Re-review complete — PR #86

**Verdict: APPROVED**

### What changed since last review (14:22 UTC)

New commit `fix(ci): file-based script timeout for emulator-runner` (`a5c014b`). Refactors the Linux/macOS emulator timeout from an inline `script:` YAML block to a file-based approach using temp files, because the `reactivecircus/android-emulator-runner` action splits multi-line `script:` blocks into separate `/bin/sh -c` calls which break `if`/`while` blocks.

**Changes:**
1. New `Write emulator scripts` step creates `/tmp/emulator-script.sh` (script + cleanup) and `/tmp/emulator-timeout.sh` (standalone timeout wrapper)
2. `Emulator (Linux/macOS)` `script:` changed from inline YAML to `/tmp/emulator-timeout.sh ${{ inputs.script-timeout }}`
3. `Run script (Windows)` — unchanged from prior review

### 🟡 Non-blocking (follow-up)

1. **`.github/actions/capabilities/android-emulator/action.yml:57`** — The Linux/macOS timeout wrapper is missing the `script-timeout=0` (no limit) guard that the Windows version has (lines 153–164). When `script-timeout=0`, the Windows step correctly skips the loop, but the Linux/macOS wrapper evaluates `[ 0 -ge 0 ]` true on the first iteration and kills immediately. Inline suggestion posted in review.

### Previously reviewed (unchanged, still correct)

All 11 other files unchanged from prior review rounds:
- `CHANGELOG.md` / `CHANGELOG.pre.md` — 1.0.6 entries
- `tool/compile_rust.sh` — libbinaryen shared library
- `.github/actions/capabilities/fvm/action.yml`, `gradle-cache/action.yml`, `pods-cache/action.yml`, `rust/action.yml`, `wasm-cache/action.yml`, `xcode-cache/action.yml` — cache key `runner.arch` additions
- `.github/workflows/flutter-upgrade.yml` — JSON parsing resilience
- `.github/workflows/full-test.yml` — timeout 40 → 60 for macos-intel

**Review by slopfairy — COMMENTED** [review]
Reviewing new commit: file-based script timeout for emulator-runner

File: .github/actions/capabilities/android-emulator/action.yml [hunk 4]
```diff
@@ -35,6 +39,40 @@ runs:
...
+      shell: bash
+      run: |
+        cat > /tmp/emulator-script.sh << 'EMUSCRIPT'
+        ${{ inputs.script }}
+        adb -s emulator-5554 emu kill 2>/dev/null; true
+        pkill -f 'crashpad[_]handler' 2>/dev/null; true
+        EMUSCRIPT
+        cat > /tmp/emulator-timeout.sh << 'EMUWRAP'
+        #!/usr/bin/env bash
+        LIMIT=${1:-2700}
```
Comment (slopfairy) [inline .github/actions/capabilities/android-emulator/action.yml]: The Linux/macOS timeout wrapper is missing the `script-timeout=0` (no limit) guard that the Windows version has. When `script-timeout` is `0`, the Windows step correctly skips the timeout loop (`[ "$LIMIT" -gt 0 ]` is false), but the Linux/macOS wrapper evaluates `[ 0 -ge 0 ]` as true on the first iteration and immediately kills the script.

Suggest adding the same guard:

```bash
LIMIT=${1:-2700}
if [ "$LIMIT" -gt 0 ] 2>/dev/null; then
  bash /tmp/emulator-script.sh &
  PID=$!
  …
  wait $PID 2>/dev/null
  exit $?
else
  bash /tmp/emulator-script.sh
  exit $?
fi
```

Not a merge blocker — `script-timeout` defaults to `2700` in practice — but worth aligning for correctness.

**Review by slopfairy — DISMISSED** [review]
## Verdict: APPROVE

One new commit since last review: `fix(ci): file-based script timeout for emulator-runner` (`a5c014b`).

### What changed

The Linux/macOS emulator timeout was refactored from an inline `script:` block to a file-based approach:

1. **New step `Write emulator scripts`** — Writes two temp files:
   - `/tmp/emulator-script.sh` — contains the user's script + cleanup (adb kill, pkill crashpad)
   - `/tmp/emulator-timeout.sh` — standalone timeout wrapper with the same 5s-polling logic as before

2. **`Emulator (Linux/macOS)`** — Changed from `script: |` multiline YAML block to `script: /tmp/emulator-timeout.sh ${{ inputs.script-timeout }}`

3. **`Run script (Windows)`** — Unchanged from prior review (inline timeout).

**Motivation**: The `reactivecircus/android-emulator-runner` action splits multi-line `script:` blocks into separate `/bin/sh -c` calls, which breaks `if`/`while` blocks. A single-file command avoids that entirely.

### Verdict on changes

The file-based approach is well-justified and correctly implemented. Heredocs use quoted delimiters (`'EMUSCRIPT'`, `'EMUWRAP'`) so bash variables in the timeout wrapper are written literally. File permissions are set with `chmod +x`. Edge cases (process exits before loop, race on `kill -9`) handled the same way as the Windows variant.

### 🟡 Non-blocking (follow-up)

1. **`.github/actions/capabilities/android-emulator/action.yml:57`** — The Linux/macOS timeout wrapper is missing the `script-timeout=0` (no limit) guard that the Windows version has (lines 153–164). With `script-timeout=0`, the Windows step correctly skips the loop, but the Linux/macOS wrapper evaluates `[ 0 -ge 0 ]` true on the first iteration and kills immediately. Suggested fix posted inline.

### Previously reviewed (unchanged, still correct)

- `CHANGELOG.md` / `CHANGELOG.pre.md` — 1.0.6 entries
- `tool/compile_rust.sh` — libbinaryen shared library
- `.github/actions/capabilities/fvm/action.yml` — pub cache `runner.arch`
- `.github/actions/capabilities/gradle-cache/action.yml` — Gradle cache `runner.os` + `runner.arch`
- `.github/actions/capabilities/pods-cache/action.yml` — CocoaPods cache `runner.os` + `runner.arch`
- `.github/actions/capabilities/rust/action.yml` — Cargo cache `runner.arch`
- `.github/actions/capabilities/wasm-cache/action.yml` — WASM cache `runner.os` + `runner.arch`
- `.github/actions/capabilities/xcode-cache/action.yml` — Xcode cache `runner.os` + `runner.arch`
- `.github/workflows/flutter-upgrade.yml` — JSON parsing resilience
- `.github/workflows/full-test.yml` — timeout 40 → 60 for macos-intel

**Comment by slopfairy** [comment]
## Re-review complete — PR #86

**Verdict: APPROVED**

### What changed since last review (14:45 UTC)

New commit `34529218` — `fix(ci): guard script-timeout=0 in Linux/macOS wrapper`

### 🟡 Previously flagged item — now fixed

1. **`.github/actions/capabilities/android-emulator/action.yml`** — The missing `script-timeout=0` guard I flagged in the previous review is now addressed. The Linux/macOS timeout wrapper uses `if [ "$LIMIT" -gt 0 ] 2>/dev/null` to skip the background/poll loop when `LIMIT=0`, falling through to run the script directly. This correctly aligns with the Windows path which already had this guard.

### ✅ All previously reviewed files remain unchanged and correct

All 11 other files from prior review rounds are unchanged:
- `CHANGELOG.md` / `CHANGELOG.pre.md` — 1.0.6 entries
- `tool/compile_rust.sh` — libbinaryen shared library
- Cache key files (fvm, gradle-cache, pods-cache, rust, wasm-cache, xcode-cache) — `runner.arch` additions
- `.github/workflows/flutter-upgrade.yml` — JSON parsing resilience
- `.github/workflows/full-test.yml` — timeout 40 → 60 for macos-intel

**Actions taken:**
- Reviewed new commit diff
- Resolved the review thread for the `script-timeout=0` guard (now fixed)
- Submitted review verdict: **APPROVE**

**Review by slopfairy — DISMISSED** [review]
## Verdict: APPROVE

### What changed since last review (14:45 UTC)

New commit `34529218` — `fix(ci): guard script-timeout=0 in Linux/macOS wrapper`

### 🟡 Previously flagged item — now fixed

1. **`.github/actions/capabilities/android-emulator/action.yml`** — The missing `script-timeout=0` guard I flagged in the previous review is now addressed. The Linux/macOS timeout wrapper uses `if [ "$LIMIT" -gt 0 ] 2>/dev/null` to skip the background/poll loop when `LIMIT=0`, falling through to `bash /tmp/emulator-script.sh` directly. Correct alignment with the Windows path.

### ✅ All previously reviewed files remain unchanged and correct

All 11 other files from prior review rounds (changelogs, cache keys, flutter-upgrade, compile_rust, full-test.yml) have been re-reviewed in prior rounds and are unchanged.

**Comment by slopfairy** [comment]
## Re-review complete — PR #86

**Verdict: APPROVED**

### What changed since last review (14:48 UTC)

New commit `b69e263` — `refactor(ci): single matrix, tight timeouts, CI timeout multiplier` by @chaudharydeepanshu (264 additions, 391 deletions, 23 files).

### Changes reviewed — NEW

**1. `Makefile` — CI timeout multiplier**
Added `TIMEOUT := $(if $(CI),--timeout=30x,)`. All `$(DART) test` invocations now pass `--timeout=30x` in CI. Local dev gets an empty string (native test timeouts apply). This allows tightening local per-test timeouts to 1s/3s while giving CI 30s/90s headroom. Sound approach.

**2. `test/ops/core/*_test.dart` (6 files) + `test/ops/stress/*_test.dart` (6 files) — Tightened timeouts**
All core tests changed from `Duration(seconds: 30)` to `Duration(seconds: 1)`. Multi-page and encryption tests changed to `Duration(seconds: 3)`. Stress tests changed from various values to `Duration(seconds: 3)`. With the 30x CI multiplier, effective CI timeouts are 30s for core and 90s for stress — adequate for CI variability.

**3. `.github/actions/capabilities/android-emulator/action.yml` — Major simplification**
Removed the entire custom script-timeout wrapper system (PID tracking, background processes, temp files, `kill -9`, etc.) that accumulated over the previous 6 commits. Now passes `script: ${{ inputs.script }}` directly to `reactivecircus/android-emulator-runner@v2` which handles execution natively. Net -63 lines. The `⚠ on macos-intel emulator` warning is preserved in the workflow matrix comment.

**4. `.github/workflows/full-test.yml` — Single matrix merge**
Core + portability rows merged into one sorted list. Portability rows marked `[P]` with `portability: true` field, skipped when toggle is off. `Check filter` step handles portability + `FILTER` logic. `fail-fast: false` prevents cross-job cancellation. Gate evaluates `needs.test.result`.

**5. `.gitignore` — Added `/test-results/`**
Standard addition for test result artifacts.

**6. `docs/ARCHITECTURE.md` — §5 updated for single matrix layout**
Comprehensive documentation of the new matrix structure, including vocabulary table, principles, capabilities list, workflow table, and test matrix table.

### No issues found

All changes are well-reasoned, internally consistent, and correctly implemented. The CI timeout multiplier approach is clean and addresses the root problem (slow CI without slowing local feedback). The emulator simplification is a positive refactoring after the iterative complexity of the prior approach.

### Previously reviewed (unchanged, still correct)

- `CHANGELOG.md` / `CHANGELOG.pre.md` — 1.0.6 entries
- `tool/compile_rust.sh` — libbinaryen shared library
- All cache key files — `runner.arch` additions
- `.github/workflows/flutter-upgrade.yml` — JSON parsing resilience

### Actions taken

- [x] Reviewed 23 file diffs from new commit `b69e263`
- [x] Read all core test, stress test, workflow, and config files on the PR HEAD
- [x] Submitted review verdict: **APPROVE**

**Review by slopfairy — APPROVED** [review]
## Verdict: APPROVE

New commit `b69e263` refactors the CI matrix into a single sorted list, adds a Makefile-based CI timeout multiplier (`--timeout=30x`), tightens all test timeouts (1s core / 3s stress), and simplifies the Android emulator action by removing the custom script-timeout wrapper in favor of the upstream action's native handling.

All changes are well-reasoned and correctly implemented. No issues found.
